src/java.base/share/classes/java/lang/classfile/AnnotationValue.java

```diff
@@ -662,6 +678,7 @@ static OfArray ofArray(AnnotationValue... values) {
      *         an enum constant, or an array of one of these.
      */
     static AnnotationValue of(Object value) {
+        requireNonNull(value);
```

Comment:
Below is a null test throwing IAE.

Reply:
The test fails if the method does not throw an exception or throws anything other than a NPE. That's why I added `requireNonNull` here.

What exceptions would be considered ok when nulls are provided?

Reply:
I think we should prefer NPE instead of IAE for null inputs.

I think we should just remove this method - the format for composite list/array or annotation is not well-defined. We might replace this with a `ofResolvedConstant` to accept primitives and String.

Reply:
API method removal is not subject of this bug/PR, please discuss it on the mailing list.
Also throwing NPE instead of IAE is a subject for consideration, as it is a factory of AnnotationValue from an Object, where null might be considered as IAE.